src/Microsoft.AspNetCore.Blazor/RenderTree/RenderTreeBuilder.cs

```diff
@@ -141,14 +149,44 @@ public void AddAttribute(int sequence, string name, string value)
         }
 
         /// <summary>
-        /// Appends a frame representing an <see cref="UIEventArgs"/>-valued attribute.
+        /// <para>
+        /// Appends a frame representing an <see cref="UIEventHandler"/>-valued attribute.
+        /// </para>
+        /// <para>
         /// The attribute is associated with the most recently added element. If the value is <c>null</c> and the
         /// current element is not a component, the frame will be omitted.
+        /// </para>
         /// </summary>
+        /// <param name="builder">The <see cref="RenderTreeBuilder"/>.</param>
         /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
         /// <param name="name">The name of the attribute.</param>
         /// <param name="value">The value of the attribute.</param>
         public void AddAttribute(int sequence, string name, UIEventHandler value)
+        {
+            AddAttribute(sequence, name, (MulticastDelegate)value);
+        }
+
+        /// <summary>
+        /// <para>
+        /// Appends a frame representing a delegate-valued attribute.
+        /// </para>
+        /// <para>
+        /// The attribute is associated with the most recently added element. If the value is <c>null</c> and the
+        /// current element is not a component, the frame will be omitted.
+        /// </para>
+        /// </summary>
+        /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
+        /// <param name="name">The name of the attribute.</param>
+        /// <param name="value">The value of the attribute.</param>
+        /// <remarks>
+        /// This method is provided for infrastructure purposes, and is used to be
+        /// <see cref="UIEventHandlerRenderTreeBuilderExtensions"/> to provide support for delegates of specific
+        /// types. For a good programming experience when using a custom delegate type, define an
+        /// extension method similar to 
+        /// <see cref="UIEventHandlerRenderTreeBuilderExtensions.AddAttribute(RenderTreeBuilder, int, string, UIChangeEventHandler)"/>
+        /// that calls this method.
+        /// </remarks>
+        public void AddAttribute(int sequence, string name, MulticastDelegate value)
```

Comment:
See the commit message for details. Tuning the overload set of of `AddAttribute` to make it do everything we want was a little tricky.